lisa/sut_orchestrator/azure/common.py

```diff
@@ -1904,3 +1915,278 @@ def get_size(disk_type: schema.DiskType, data_disk_iops: int = 1) -> int:
             return iops_dict[min_iops]
         else:
             raise LisaException(f"Data disk type {disk_type} is unsupported.")
+
+
+def get_certificate_client(
+    vault_url: str, platform: "AzurePlatform"
+) -> CertificateClient:
+    return CertificateClient(vault_url, platform.credential)
+
+
+def get_secret_client(vault_url: str, platform: "AzurePlatform") -> SecretClient:
+    return SecretClient(vault_url, platform.credential)
+
+
+def get_key_vault_management_client(
+    platform: "AzurePlatform",
+) -> KeyVaultManagementClient:
+    return KeyVaultManagementClient(platform.credential, platform.subscription_id)
+
+
+def get_tenant_id(credential: Any) -> Any:
+    # Initialize the Subscription client
+    subscription_client = SubscriptionClient(credential)
+    # Get the subscription
+    subscription = next(subscription_client.subscriptions.list())
+    return subscription.tenant_id
+
+
+def get_identity_id(
+    platform: "AzurePlatform", application_id: Optional[str] = None
+) -> Any:
+    # If application_id is not provided or is None, use /me endpoint
+    if not application_id:
+        base_url = "https://graph.microsoft.com/"
+        api_version = "v1.0"
+        endpoint = "me"
+        graph_api_url = f"{base_url}{api_version}/{endpoint}"
+    else:
+        base_url = "https://graph.microsoft.com/"
+        api_version = "v1.0"
+        endpoint = f"servicePrincipals(appId='{application_id}')"
+        graph_api_url = f"{base_url}{api_version}/{endpoint}"
```

Comment:
move the common code above if

    base_url = "https://graph.microsoft.com/"
    api_version = "v1.0"
    if not application_id:
        endpoint = "me"
    else:
        endpoint = f"servicePrincipals(appId='{application_id}')"
    graph_api_url = f"{base_url}{api_version}/{endpoint}"

Reply:
this makes better thanks